src/infra/core/src/engine/engine_spark.rs

```diff
@@ -74,15 +74,32 @@ impl SparkEngine {
 
     async fn materialize_request(
         &self,
+        dataset: &dyn Dataset,
         request: IngestRequest,
         operation_dir: &Path,
     ) -> Result<MaterializedEngineRequest, InternalError> {
-        let dataset = self
-            .dataset_repo
-            .get_dataset(&request.dataset_handle.as_local_ref())
-            .await
-            .int_err()?;
+        // TODO: This seems dirty
+        let some_random_hash: Multihash = Multihash::from_digest_sha3_256(b"");
+        let data_url = dataset
+            .as_data_repo()
+            .get_internal_url(&some_random_hash)
+            .await;
+
+        if data_url.scheme() == "file" {
```

Comment:
Can we have some enum or set of constants for schemes defined? This looks a bit weird we copy similar strings, not just `file`, but other schemes too.